src/library/autodj/autodjprocessor.cpp

```diff
@@ -419,18 +424,15 @@ AutoDJProcessor::AutoDJError AutoDJProcessor::toggleAutoDJ(bool enable) {
 
         TrackPointer nextTrack = getNextTrackFromQueue();
         if (!nextTrack) {
-            qDebug() << "Queue is empty now";
-            if (m_pEnabledAutoDJ->get() != 0.0) {
-                m_pEnabledAutoDJ->set(0.0);
-            }
+            qDebug() << "Queue is empty now, disable Auto DJ";
+            m_pEnabledAutoDJ->setAndConfirm(0.0);
```

Comment:
I think we don't need these in the `enable` anymore.
Didn't look close enough, yet, to be sure.